server/gateway.go

```diff
@@ -1575,17 +1791,13 @@ func (s *Server) endSubjectNoInterestForGateways(accName string, sub *subscripti
 	}
 }
 
-// This is invoked when a queue subscription is registered.
-func (s *Server) sendQueueSubToGateways(accName string, sub *subscription, isRemote bool) {
-	if sub.queue == nil {
+// This is invoked when the first (or last) queue subscription on a
+// given subject/group is registered (or unregistered). Sent to all
+// inbound gateways.
+func (s *Server) sendQueueSubOrUnsubToGateways(accName string, qsub *subscription, added bool) {
+	if qsub.queue == nil {
```

Comment:
Should this now be combined with plain sub version? And the if plainsub and not sas (or mode=PERFECT) skip?

Reply:
No. For plain subs, code is a bit more involved when in OPTIMISTIC mode, which we don't have to bother with queues. I would really like to keep them separate for now.

Reply:
ok